### PR TITLE
CASSANDRA-19128

### PR DESCRIPTION
**Pull request by ifesdjeen**

### CODE REVIEW

File: src/java/org/apache/cassandra/tcm/transformations/PrepareMove.java [hunk 1]
```diff
@@ -67,6 +67,11 @@ public PrepareMove(NodeId nodeId,
         this.streamData = streamData;
     }
 
+    public NodeId nodeId()
```
Comment (krummas) [inline src/java/org/apache/cassandra/tcm/transformations/PrepareMove.java:71]: maybe mark VisibleForTesting

File: src/java/org/apache/cassandra/tcm/MetadataKeys.java [hunk 2]
```diff
@@ -52,4 +60,44 @@ public static MetadataKey make(String...parts)
...
     }
 
+    public static ImmutableSet<MetadataKey> diffKeys(ClusterMetadata before, ClusterMetadata after)
+    {
+        ImmutableSet.Builder<MetadataKey> builder = new ImmutableSet.Builder<>();
+        diffKeys(before, after, builder::add);
+        return builder.build();
+    }
+
+    private static void diffKeys(ClusterMetadata before, ClusterMetadata after, Consumer<MetadataKey> consumer)
```
Comment (krummas) [inline src/java/org/apache/cassandra/tcm/MetadataKeys.java]: this `Consumer` is always `builder::add`, could use that directly above instead (and move the logic in this helper method there)

**Review by krummas — COMMENTED** [review]
2 nits inline

File: src/java/org/apache/cassandra/tcm/MetadataKeys.java (hunk 2 above)
Comment (ifesdjeen) [reply src/java/org/apache/cassandra/tcm/MetadataKeys.java]: I could yeah; i just had an follow-up patch that would also add checks for cluster metadata equality / epoch bump, but I guess i can separate the two without pre-committing any code

**Pull request closed without merging**